app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt

```diff
@@ -173,43 +168,49 @@ private fun MessageStatusLabel(messageStatus: MessageStatus) {
-                        vertical = dimensions().spacing2x
+        when(messageStatus) {
+            MessageStatus.Deleted,
+            is MessageStatus.Edited,
+            MessageStatus.ReceiveFailure -> {
+                Box(
+                    modifier = Modifier
+                        .wrapContentSize()
+                        .border(
+                            BorderStroke(
+                                width = 1.dp,
+                                color = MaterialTheme.wireColorScheme.divider
+                            ),
+                            shape = RoundedCornerShape(size = dimensions().spacing4x)
+                        )
+                        .padding(
+                            horizontal = dimensions().spacing4x,
+                            vertical = dimensions().spacing2x
+                        )
+                ) {
+                    Text(
+                        text = messageStatus.text.asString(),
+                        style = LocalTextStyle.current.copy(color = MaterialTheme.wireColorScheme.labelText)
                     )
-            ) {
-                Text(
-                    text = stringResource(id = messageStatus.stringResourceId),
-                    style = LocalTextStyle.current.copy(color = MaterialTheme.wireColorScheme.labelText)
-                )
+                }
             }
-        } else {
-            Row {
-                Text(
-                    text = stringResource(id = messageStatus.stringResourceId),
-                    style = LocalTextStyle.current.copy(color = MaterialTheme.colorScheme.error)
-                )
-                Spacer(Modifier.width(dimensions().spacing4x))
-                Text(
-                    modifier = Modifier.fillMaxWidth(),
-                    style = LocalTextStyle.current.copy(
-                        color = MaterialTheme.wireColorScheme.onTertiaryButtonSelected,
-                        textDecoration = TextDecoration.Underline
-                    ),
-                    text = stringResource(R.string.label_try_again),
-                )
+            MessageStatus.SendFailure -> {
+                Row {
+                    Text(
+                        text = messageStatus.text.asString(),
+                        style = LocalTextStyle.current.copy(color = MaterialTheme.colorScheme.error)
+                    )
+                    Spacer(Modifier.width(dimensions().spacing4x))
+                    Text(
+                        modifier = Modifier.fillMaxWidth(),
+                        style = LocalTextStyle.current.copy(
+                            color = MaterialTheme.wireColorScheme.onTertiaryButtonSelected,
+                            textDecoration = TextDecoration.Underline
+                        ),
+                        text = stringResource(R.string.label_try_again),
+                    )
+                }
             }
+            else -> {}
```

Comment:
Wouldn't it be better to be exhaustive here and add `MessageStatus.Untouched` instead of `else`??

Reply:
You're right, I changed it 🚀